doc/source/whatsnew/v1.4.2.rst

```diff
@@ -17,6 +17,7 @@ Fixed regressions
 - Fixed regression in :meth:`DataFrame.drop` and :meth:`Series.drop` when :class:`Index` had extension dtype and duplicates (:issue:`45860`)
 - Fixed memory performance regression in :meth:`Series.fillna` when called on a :class:`DataFrame` column with ``inplace=True`` (:issue:`46149`)
 - Provided an alternative solution for passing custom Excel formats in :meth:`.Styler.to_excel`, which was a regression based on stricter CSS validation. Examples available in the documentation for :meth:`.Styler.format` (:issue:`46152`)
+- Fixed regression in :meth:`DataFrame.loc.__setitem__` losing :class:`MultiIndex` names if :class:`DataFrame`  was empty before (:issue:`46317`)
```

Comment:
:meth:`DataFrame.loc.__setitem__` not a valid ref. will change in a release note tidy up

Reply:
What would be valid here? (For future references)

Reply:
in 1.4.1 release notes https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.4.1.html#fixed-regressions , I changed it to

> Regression when setting values with [DataFrame.loc()](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.loc.html#pandas.DataFrame.loc) losing [Index](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.Index.html#pandas.Index) name if [DataFrame](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.html#pandas.DataFrame) was empty before ([GH45621](https://github.com/pandas-dev/pandas/issues/45621))

will probably do similar

Reply:
done in #46354